pandas/core/indexes/datetimelike.py

```diff
@@ -130,11 +134,11 @@ def _ndarray_values(self):
     # Abstract data attributes
 
     @property
```

Comment:
why this change?

Reply:
mypy has the following error for the annotated value function: 

`pandas/core/indexes/datetimelike.py:139: error: "None" has no attribute "_data"`

I add annotation ` # type: DatetimeArray ` to `_data`. But mypy still has the same error.
https://github.com/pandas-dev/pandas/blob/8154efb0c1a64295cf54e00025b4ab09bcd02752/pandas/core/indexes/datetimes.py#L327

I guess that mypy mixed up the instance attribute `_data` with the class attributes `_data` which is set `None` in the beginning of DatetimeIndexOpsMixin.

Reply:
cc @WillAyd

Reply:
Sorry missed this ping. This could be a similar conversation to https://github.com/pandas-dev/pandas/pull/26518/files#r287573810

Reply:
I think a better solution is to provide the appropriate type to `_data`. Can you create a `DatetimeLikeArray` in pandas._typing which should be something like `DatetimeLikeArray = TypeVar('DatetimeLikeArray', DatetimeArray, PeriodArray, TimedeltaArray)` and assign that type to `_data` in the class? I think that should resolve.

cc @jbrockmendel in case he has other insights on the types here